src/main-process/atom-application.js

```diff
@@ -452,13 +452,17 @@ class AtomApplication extends EventEmitter {
 
   // Registers basic application commands, non-idempotent.
   handleEvents () {
-    const getLoadSettings = () => {
+    const getLoadSettings = includeWindow => {
       const window = this.focusedWindow()
-      return {devMode: window && window.devMode, safeMode: window && window.safeMode}
+      return {
+        devMode: window && window.devMode,
+        safeMode: window && window.safeMode,
+        window: includeWindow && window
+      }
     }
 
     this.on('application:quit', () => app.quit())
-    this.on('application:new-window', () => this.openPath(getLoadSettings()))
+    this.on('application:new-window', () => this.openPath(getLoadSettings(false)))
```

Comment:
This isn't something that should hold up this pull request, but I think calls to `getLoadSettings` could be more intention-revealing/self-documenting we used a named argument. 

```js
getLoadSettings({includeWindow: false})
```

What do you think?

Reply:
Yah, that's reasonable. Honestly this whole class could use an overhaul to its readability. I tried to at least leave comments where I was changing things.